homeassistant/components/conversation/default_agent.py

```diff
@@ -118,25 +75,119 @@ async def async_process(
         language: str | None = None,
     ) -> ConversationResult | None:
```

Comment:
Do you remember the use case of being able to return `None` here ?

Reply:
It signals that the match was not successful. With how hassil works right now, you can't tell if the problem is an incorrect slot value or if the sentence wouldn't match any intent.

Reply:
I think that it makes sense to always try to return errors as conversation results. Since sometimes the conversation result already is an error. And then we can try to be specific in errors where possible (ie language not supported).